src/libponyrt/actor/actor.c

```diff
@@ -589,3 +656,45 @@ PONY_API void pony_poll(pony_ctx_t* ctx)
   pony_assert(ctx->current != NULL);
   ponyint_actor_run(ctx, ctx->current, 1);
 }
+
+void ponyint_actor_setoverloaded(pony_actor_t* actor)
+{
+  set_flag(actor, FLAG_OVERLOADED);
+}
+
+bool ponyint_actor_overloaded(pony_actor_t* actor)
+{
+  return has_flag(actor, FLAG_OVERLOADED);
+}
+
+void ponyint_actor_unsetoverloaded(pony_ctx_t* ctx, pony_actor_t* actor)
+{
+  unset_flag(actor, FLAG_OVERLOADED);
+  if (!has_flag(actor, FLAG_UNDER_PRESSURE))
+    ponyint_sched_unmute_senders(ctx, actor, true);
+}
+
+PONY_API void pony_apply_backpressure()
+{
+  set_flag(pony_ctx()->current, FLAG_UNDER_PRESSURE);
+}
+
+PONY_API void pony_release_backpressure()
+{
+  pony_ctx_t* ctx = pony_ctx();
+  unset_flag(ctx->current, FLAG_UNDER_PRESSURE);
+  if (!has_flag(ctx->current, FLAG_OVERLOADED))
+      ponyint_sched_unmute_senders(ctx, ctx->current, true);
+}
+
+bool ponyint_triggers_muting(pony_actor_t* actor)
```

Comment:
Same as above.

Reply:
Given this only needs the actor it’s unclear to me why we should do that

Reply:
That's true, I missed that.